src/components/fields/MultiSchemaField.js

```diff
@@ -21,17 +22,24 @@ class AnyOfField extends Component {
     };
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const matchingOption = this.getMatchingOption(
-      nextProps.formData,
-      nextProps.options
-    );
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      !deepEquals(this.props.formData, prevProps.formData) &&
+      this.props.idSchema.$id === prevProps.idSchema.$id
+    ) {
+      const matchingOption = this.getMatchingOption(
+        this.props.formData,
+        this.props.options
+      );
 
-    if (matchingOption === this.state.selectedOption) {
-      return;
-    }
+      if (!prevState || matchingOption === this.state.selectedOption) {
```

Comment:
Why did you add the `!prevState` check @SuriGill ? It seems like the tests already pass without it.